api/handler/copy.go

```diff
@@ -111,6 +111,14 @@ func (h *handler) CopyObjectHandler(w http.ResponseWriter, r *http.Request) {
 		}
 	}
 
+	if isBucketOwnerPreferred(eacl.EACL) {
```

Comment:
Why doing it for `!containsACL`?

Reply:
`containsACL` means the user has passed some ACL headers. This check is not related to passed parameters. This check is about the actual ACL and ownership mode

Reply:
But if there are no ACLs passed we can just do whatever is appropriate in the default case, no?

Reply:
I think in the `BucketOwnerPreferred` mode, the header `bucket-owner-full-control` must be set

Reply:
I hope @evgeniiz321 will figure it out.